client/js/libs/handlebars/ircmessageparser/parseName.js

```diff
@@ -22,31 +20,26 @@ module.exports = function(orig) {
 
 // Finds and stores the suffix / prefix to nicks, like @ and : Otherwise the URL parsed would be correct but the color generated would be wrong.
 function replaceAndKeep(orig) {
-	let prefix = orig[0];
-	let suffix = orig[orig.length - 1];
+	let prefix = "";
+	let suffix = "";
 	let cleaned = orig;
 
 	// TODO: User modes should be RPL_ISUPPORT.PREFIX
-	if (prefix === "@" || prefix === "+" || prefix === "%" || prefix === "!") {
-		cleaned = cleaned.substring(1, cleaned.length);
-	} else {
-		prefix = "";
-	}
-
-	if (suffix === "," || suffix === ":" || suffix === "s") {
-		let strip = 1; // amount of characters to strip
-
-		if (suffix === "s") {
-			if (orig[orig.length - 2] === "'") { // ends with 's
-				suffix = "'s";
-				strip++;
-			}
+	const prefixes = ["@", "+", "%", "!"];
+	prefixes.map((e) => {
+		if (orig.startsWith(e)) {
+			cleaned = cleaned.substring(e.length, cleaned.length);
+			prefix = e;
 		}
+	});
```

Comment:
This prefix and suffix logic could be simplified in another way, without multiple `startsWith` checks:

```
const prefixes = ["@", "+", "%", "!"];

let prefix = "";
let matchedPrefixPos = prefixes.indexOf(orig[0]);

if (matchedPrefixPos !== -1) {
    prefix = prefixes[matchedPrefixPos];
    cleaned = cleaned.substring(0, cleaned.length - prefix.length);
}
```

Reply:
won't work if the suffix/prefix we're looking for is length > 1 (due to `indexOf(orig[0])`)